app/services/invoices/refresh_draft_service.rb

```diff
@@ -17,24 +18,39 @@ def call
       return result unless invoice.draft?
 
       ActiveRecord::Base.transaction do
-        invoice.credit_notes.destroy_all
-        invoice.credits.destroy_all
-        invoice.wallet_transactions.destroy_all
+        cn_subscription_ids = invoice.credit_notes.map do |cn|
+          { credit_note_id: cn.id, subscription_id: cn.fees.pick(:subscription_id) }
+        end
+        invoice.credit_notes.each { |cn| cn.items.update_all(fee_id: nil) }
```

Comment:
Rails/SkipsModelValidations: Avoid using `update_all` because it skips validations. (https://guides.rubyonrails.org/active_record_validations.html#skipping-validations)